src/App.js

```diff
@@ -26,14 +26,15 @@ function App() {
         token
           ? db.collection(token).onSnapshot((snapshot) => {
               const newList = [];
+              // Restructure data from array to an array of objects,
+              // with each object containing name, id and date of the last purchase
               snapshot.forEach((doc) => {
-                // This seems to work, while snapshot.map doesn't
-                newList.push(
-                  JSON.stringify(doc.data()['formData']['itemName']).replace(
-                    /['"]+/g,
-                    '',
-                  ),
-                );
+                const obj = {
+                  id: doc.id,
+                  itemName: doc.data()['formData']['itemName'],
+                  lastPurchaseDate: doc.data()['formData']['lastPurchaseDate'],
+                };
```

Comment:
🚀

Reply:
🤯  Nice job!! And the comments here are super helpful, it would have taken me a very long time to understand what was happening otherwise!